snmp/datadog_checks/snmp/snmp.py

```diff
@@ -329,13 +332,36 @@ def _consume_binds_iterator(self, binds_iterator, config):
             all_binds.extend(var_binds_table)
         return all_binds, error
 
+    def _start_discovery(self):
+        cache = read_persistent_cache(self.check_id)
+        if cache:
+            hosts = json.loads(cache)
+            for host in hosts:
+                try:
+                    ipaddress.ip_address(host)
+                except ValueError:
+                    write_persistent_cache(self.check_id, json.dumps([]))
+                    break
+                instance = self.instance.copy()
+                instance.pop('network_address')
+                instance['ip_address'] = host
+
+                host_config = self._build_config(instance)
+                self._config.discovered_instances[host] = host_config
```

Comment:
1) Do we need here to also call `host_config.refresh_with_profile` ? 

2) If 1) yes. Seems the logic is similar to, maybe we can factor:
https://github.com/DataDog/integrations-core/blob/0d0efa0182a3bcc19c264f728429ab7cf7de5b1b/snmp/datadog_checks/snmp/snmp.py#L138-L154

Reply:
No that'll be done by `check` in the first run.

Reply:
Maybe those 3 lines, from this related part, are not necessary then: https://github.com/DataDog/integrations-core/blob/0d0efa0182a3bcc19c264f728429ab7cf7de5b1b/snmp/datadog_checks/snmp/snmp.py#L152-L154

But note fully related to this PR.